### ISSUE TITLE
Multiple and Shared Instances?

### ISSUE THREAD

**Issue by codemann8** [sansiaali/mulanatrax#2]
~~So if I build out a whole series of maps for a few hours, then shut down the computer because I don't think anyone can do a first playthru in one sitting, that all progress is lost for next time? Or am I missing a step, it seems every time I boot up this program it is blank. Does it not save or retain any information?~~

Awesome program otherwise.

Do you think it would be possible to somehow make it so you can share sessions across browsers/clients? Store it on the server side somewhere. I suppose you'd have to have some way to delete/reset instances, or change to another instance if so. But would be nice.

Another neat feature if the above is possible, to have a read-only view that could be shared with viewers on Twitch or other live streaming.

Ik you're a few years separated from coding this, so I'm sure you're just done with this, but this tool could def have some more universal applications beyond just LM.

**Comment by codemann8** [comment]
Sorry, first image I added went away after closing, but seems to save now, idk what I did different. This program is crazy awesome.

**Comment by sansiaali** [comment]
Nice to hear you got it working. The browser's IndexedDB/localstorage is used for saving your progress. Thanks for the kind words

**Comment by codemann8** [comment]
@sansiaali Yeah, I think that's why it 'disappeared' on me. I initially tried it locally, but then I changed package.json to call vise with --host so that I could run the browser from my tablet instead. Speaking of, gonna reopen this and edit the OP.

**Comment by sansiaali** [comment]
As the data is stored in IndexedDB, it is bound to the specific browser that you use the application with.

The database is used with a library called dexie which focuses on IndexedDB so serverside stuff isn't feasible without a rewrite of some sort. You could rewrite dexie queries into sqlite and host the data in a server. Adding public facing stuff would require authorization etc.

I created this for myself while playing la-mulana 1/2 but the code could be forked and adapted to suit other games pretty easily I think.
